.github/workflows/pr-go-workspace-check.yml

```diff
@@ -0,0 +1,47 @@
+name: "Go Workspace Check"
+
+on:
+  workflow_dispatch:
+  pull_request:
+    branches: [main]
+    paths:
+      - '**/go.work'
+      - '**/go.work.sum'
+      - '**/go.mod'
+      - '**/go.sum'
+
+jobs:
+  check:
+    name: Go Workspace Check
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v4
+      with:
+        fetch-depth: 0
+
+    - name: Set go version
+      uses: actions/setup-go@v4
+      with:
+        go-version: '1.21.8'
```

Comment:
1.21.9 is already in main

Reply:
if there's a later PR in the preview branch that bumps this further it'll be included later and bumped, otherwise it'll stay on 1.21.8 on this branch